website/auth_pages.py

```diff
@@ -209,6 +211,65 @@ def signup(self):
         remember_current_user(user)
         return resp
 
+    @route("/verify", methods=["GET"])
+    def verify_email(self):
+        username = request.args.get("username", None)
+        token = request.args.get("token", None)
+        if not token:
+            return make_response(gettext("token_invalid"), 400)
+        if not username:
+            return make_response(gettext("username_invalid"), 400)
+
+        # Verify that user actually exists
+        user = self.db.user_by_username(username)
+        if not user:
+            return make_response(gettext("username_invalid"), 403)
+
+        # If user is already verified -> re-direct to landing-page anyway
+        if "verification_pending" not in user:
+            return redirect("/landing-page")
+
+        # Verify the token
+        if token != user["verification_pending"]:
+            return make_response(gettext("token_invalid"), 403)
+
+        # Remove the token from the user
+        self.db.update_user(username, {"verification_pending": None})
+
+        # We automatically login the user
+        cookie = make_salt()
+        self.db.store_token({"id": cookie, "username": user["username"], "ttl": times() + SESSION_LENGTH})
+        remember_current_user(user)
+
+        return redirect("/landing-page")
+
+    @route("/turn-into-teacher", methods=['POST'])
+    def turn_into_teacher_account(self):
+        username = current_user()['username']
+        if not username:
+            return make_response(gettext("username_invalid"), 400)
+        user = self.db.user_by_username(username)
+        if not user:
+            return make_response(gettext("username_invalid"), 403)
+
+        # We update the user in the database and turn it into a teacher
+        self.db.update_user(user['username'], {"is_teacher": 1, "teacher_request": None})
+
+        if user.get("email"):
+            try:
+                send_localized_email_template(
+                    locale=user["language"], template="welcome_teacher", email=user["email"], username=user["username"]
+                )
+            except Exception:
+                print(f"An error occurred when sending a welcome teacher mail to {user['email']}, "
+                      "changes still processed")
+
+        session.get('user')['is_teacher'] = True
+        session['welcome-teacher'] = True
+
+        # I want to redirect the user to a tutorial!
```

Comment:
This comment is slightly confusing. Is the teacher is sent to the For-Teachers section. Is this the goal?

Reply:
Yes you're right, it was a goal. I want to update the tutorial and send them there, I think things are stable enough now to update the teacher tutorial.